### ISSUE TITLE
Train on the ImageNet dataset throws error

### ISSUE THREAD

**Issue by MateSteinforth** [soumith/dcgan.torch#27]
OSX 10.11.5.

$ DATA_ROOT=myimages dataset=folder th main.lua
{
  ntrain : inf
  beta1 : 0.5
  name : "experiment1"
  niter : 25
  batchSize : 64
  ndf : 64
  fineSize : 64
  nz : 100
  loadSize : 96
  gpu : 1
  ngf : 64
  dataset : "folder"
  lr : 0.0002
  noise : "normal"
  nThreads : 4
  display_id : 10
  display : 1
}
Random Seed: 251	
Starting donkey with id: 1 seed: 252
table: 0x02b80fc8
Starting donkey with id: 2 seed: 253
table: 0x02ba0ca8
Starting donkey with id: 3 seed: 254
table: 0x02bc0f40
Starting donkey with id: 4 seed: 255
table: 0x02be0d88
Creating train metadata
table: 0x02ef7ee0
running "find" on each class directory, and concatenate all those filenames into a single file containing all image paths for a given class
Creating train metadata
table: 0x02bf1af0
running "find" on each class directory, and concatenate all those filenames into a single file containing all image paths for a given class
Creating train metadata
table: 0x02e00c70
running "find" on each class directory, and concatenate all those filenames into a single file containing all image paths for a given class
Creating train metadata
table: 0x02d971a8
running "find" on each class directory, and concatenate all those filenames into a single file containing all image paths for a given class
/tmp/lua_2AEOyi: line 1: gfind: command not found
/tmp/lua_9DgkXe: line 1: gfind: command not found
/tmp/lua_psrQx3: line 1: gfind: command not found
/tmp/lua_aZfoE8: line 1: gfind: command not found
now combine all the files to a single large file
now combine all the files to a single large file
now combine all the files to a single large file
now combine all the files to a single large file
load the large concatenated list of sample paths to self.imagePath
load the large concatenated list of sample paths to self.imagePath
load the large concatenated list of sample paths to self.imagePath
load the large concatenated list of sample paths to self.imagePath
sh: gwc: command not found
sh: gwc: command not found
sh: gwc: command not found
sh: gwc: command not found
/Users/WS18/torch/install/bin/luajit: /Users/WS18/torch/install/share/lua/5.1/threads/threads.lua:183: [thread 1 callback] /Users/WS18/dcgan.torch/data/dataset.lua:198: attempt to perform arithmetic on a nil value
stack traceback:
	/Users/WS18/dcgan.torch/data/dataset.lua:198: in function '__init'
	/Users/WS18/torch/install/share/lua/5.1/torch/init.lua:91: in function </Users/WS18/torch/install/share/lua/5.1/torch/init.lua:87>
	[C]: in function 'dataLoader'
	/Users/WS18/dcgan.torch/data/donkey_folder.lua:82: in main chunk
	[C]: in function 'dofile'
	/Users/WS18/dcgan.torch/data/data.lua:42: in function </Users/WS18/dcgan.torch/data/data.lua:32>
	[C]: in function 'xpcall'
	/Users/WS18/torch/install/share/lua/5.1/threads/threads.lua:234: in function 'callback'
	/Users/WS18/torch/install/share/lua/5.1/threads/queue.lua:65: in function </Users/WS18/torch/install/share/lua/5.1/threads/queue.lua:41>
	[C]: in function 'pcall'
	/Users/WS18/torch/install/share/lua/5.1/threads/queue.lua:40: in function 'dojob'
	[string "  local Queue = require 'threads.queue'..."]:13: in main chunk
stack traceback:
	[C]: in function 'error'
	/Users/WS18/torch/install/share/lua/5.1/threads/threads.lua:183: in function 'dojob'
	/Users/WS18/torch/install/share/lua/5.1/threads/threads.lua:264: in function 'synchronize'
	/Users/WS18/torch/install/share/lua/5.1/threads/threads.lua:142: in function 'specific'
	/Users/WS18/torch/install/share/lua/5.1/threads/threads.lua:125: in function 'Threads'
	/Users/WS18/dcgan.torch/data/data.lua:30: in function 'new'
	main.lua:38: in main chunk
	[C]: in function 'dofile'
	...WS18/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x0101e83d10

**Comment by MateSteinforth** [comment]
found an answer here: https://github.com/cmusatyalab/openface/issues/23#issuecomment-149175986

**Comment by MateSteinforth** [comment]
now I get this:

/Users/WS18/torch/install/bin/luajit: main.lua:45: attempt to call method 'noBias' (a nil value)
stack traceback:
	main.lua:45: in function 'callback'
	/Users/WS18/torch/install/share/lua/5.1/nn/Module.lua:318: in function 'apply'
	/Users/WS18/torch/install/share/lua/5.1/nn/Module.lua:322: in function 'apply'
	main.lua:81: in main chunk
	[C]: in function 'dofile'
	...WS18/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x0103607d10

**Comment by MateSteinforth** [comment]
fixed it by updating the stack:

luarocks install torch
luarocks install nn
luarocks install cutorch
luarocks install cunn
